securedrop/models.py

```diff
@@ -138,9 +141,9 @@ def to_json(self) -> 'Dict[str, object]':
         docs_msg_count = self.documents_messages_count()
 
         if self.last_updated:
-            last_updated = self.last_updated.isoformat() + 'Z'
+            last_updated = self.last_updated.strftime(API_DATETIME_FORMAT)
         else:
-            last_updated = datetime.datetime.utcnow().isoformat() + 'Z'
+            last_updated = datetime.datetime.utcnow().strftime(API_DATETIME_FORMAT)
```

Comment:
For the future: While it doesn't matter since we discard the TZ information, we probably should switch this to `datetime.now(timezone.utc)` for consistency.

Reply:
It's in part because this is not trivial that I'd avoid `strftime` and favor `isoformat`/`fromisoformat` instead, see #6257

Reply:
To calrify, `correct` and `wrong` are both correct **until** we encode them using the `strftime` format we defined, it's their string representation that's correct or wrong.

Reply:
The current code is correct-ish because it's using `utcnow`:
```python
from datetime import datetime, timezone

# It's 11:45 in California.
correct = datetime.now(timezone.utc)
wrong = datetime.utcnow()

API_DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
correct.strftime(API_DATETIME_FORMAT)
# '2022-02-09T07:45:00.795348Z'

wrong.strftime(API_DATETIME_FORMAT)
# '2022-02-09T07:45:26.082728Z'
```

The only distinction between `datetime.now(timezone.utc)` and `datetime.utcnow()` is what the internal tzinfo value is:
```
>>> correct.tzinfo is timezone.utc
True
>>> wrong.tzinfo is None
True
```

...which also affects how its formatted by default but not with our current strftime method.

Reply:
:bulb: Got it. I had missed that completely, thank for explaining @legoktm !